# Flight Data Clean Up

In [1]:
# Import Dependencies
import pandas as pd
from pathlib import Path
import glob
import zipfile
import os
import shutil
import re

## Extract and Load Flight Information

In [2]:
# List of all source zips
zips = ['Resources/BTS/BTS_Q1_2023.zip','Resources/BTS/BTS_Q2_2023.zip',
       'Resources/BTS/BTS_Q3_2023.zip','Resources/BTS/BTS_Q4_2023.zip']

# Directory to extract to, confirm it exists
extraction_path = 'Resources/BTS/extracted_zips'
os.makedirs(extraction_path, exist_ok=True)

# Iterate through zips and extract all
for zip in zips:
    with zipfile.ZipFile(zip, 'r') as zip_ref:
        zip_ref.extractall(extraction_path)

    print(f"Extracted {zip} to {extraction_path}")

Extracted Resources/BTS/BTS_Q1_2023.zip to Resources/BTS/extracted_zips
Extracted Resources/BTS/BTS_Q2_2023.zip to Resources/BTS/extracted_zips
Extracted Resources/BTS/BTS_Q3_2023.zip to Resources/BTS/extracted_zips
Extracted Resources/BTS/BTS_Q4_2023.zip to Resources/BTS/extracted_zips


In [3]:
# Read all extracted CSV's
# glob is an efficient way to work through all CSV's in a directory: https://docs.python.org/3/library/glob.html
csv_files = glob.glob(f"{extraction_path}/*.csv")

# Empty list for storing DataFrames
dataframes = []

# Iterate through csv's and store into DataFrames list
for csv_file in csv_files:
    df = pd.read_csv(csv_file, low_memory=False)
    dataframes.append(df)
# Concatenante into single DataFrame
unfiltered_flight_info = pd.concat(dataframes, ignore_index = True)
pd.set_option('display.max_columns', None)
unfiltered_flight_info.head()

Year  Quarter  Month  DayofMonth  DayOfWeek  FlightDate  \
0  2023        1      1          22          7  2023-01-22   
1  2023        1      1          22          7  2023-01-22   
2  2023        1      1          22          7  2023-01-22   
3  2023        1      1          22          7  2023-01-22   
4  2023        1      1          22          7  2023-01-22   

  Marketing_Airline_Network Operated_or_Branded_Code_Share_Partners  \
0                        B6                                      B6   
1                        B6                                      B6   
2                        B6                                      B6   
3                        B6                                      B6   
4                        B6                                      B6   

   DOT_ID_Marketing_Airline IATA_Code_Marketing_Airline  \
0                     20409                          B6   
1                     20409                          B6   
2                     20409                          B6   
3                     20409                          B6   
4                     20409                          B6   

   Flight_Number_Marketing_Airline Originally_Scheduled_Code_Share_Airline  \
0                             1447                                     NaN   
1                             1451                                     NaN   
2                             1453                                     NaN   
3                             1454                                     NaN   
4                             1455                                     NaN   

   DOT_ID_Originally_Scheduled_Code_Share_Airline  \
0                                             NaN   
1                                             NaN   
2                                             NaN   
3                                             NaN   
4                                             NaN   

  IATA_Code_Originally_Scheduled_Code_Share_Airline  \
0                                               NaN   
1                                               NaN   
2                                               NaN   
3                                               NaN   
4                                               NaN   

   Flight_Num_Originally_Scheduled_Code_Share_Airline Operating_Airline   \
0                                                NaN                  B6   
1                                                NaN                  B6   
2                                                NaN                  B6   
3                                                NaN                  B6   
4                                                NaN                  B6   

   DOT_ID_Operating_Airline IATA_Code_Operating_Airline Tail_Number  \
0                     20409                          B6      N636JB   
1                     20409                          B6      N3112J   
2                     20409                          B6      N507JT   
3                     20409                          B6      N956JT   
4                     20409                          B6      N249JB   

   Flight_Number_Operating_Airline  OriginAirportID  OriginAirportSeqID  \
0                             1447            12197             1219702   
1                             1451            10721             1072102   
2                             1453            11697             1169706   
3                             1454            14843             1484306   
4                             1455            10721             1072102   

   OriginCityMarketID Origin       OriginCityName OriginState  \
0               31703    HPN     White Plains, NY          NY   
1               30721    BOS           Boston, MA          MA   
2               32467    FLL  Fort Lauderdale, FL          FL   
3               34819    SJU         San Juan, PR          PR   
4               30721    BOS           Boston, MA          MA   

   

In [4]:
# Specify columns and remove limitation on display
pd.set_option('display.max_columns', None)
flights_df = pd.DataFrame(unfiltered_flight_info[['FlightDate','IATA_Code_Marketing_Airline', 
                                                  'Operated_or_Branded_Code_Share_Partners',
                                                  'Operating_Airline ','Flight_Number_Marketing_Airline',
                                                  'Origin','Dest','DepDelay','DepDelayMinutes',
                                                  'DepDel15','ArrDelay','ArrDelayMinutes','Cancelled',
                                                  'Diverted','CarrierDelay','WeatherDelay','NASDelay',
                                                  'SecurityDelay','LateAircraftDelay']])
flights_df.head()

FlightDate IATA_Code_Marketing_Airline  \
0  2023-01-22                          B6   
1  2023-01-22                          B6   
2  2023-01-22                          B6   
3  2023-01-22                          B6   
4  2023-01-22                          B6   

  Operated_or_Branded_Code_Share_Partners Operating_Airline   \
0                                      B6                 B6   
1                                      B6                 B6   
2                                      B6                 B6   
3                                      B6                 B6   
4                                      B6                 B6   

   Flight_Number_Marketing_Airline Origin Dest  DepDelay  DepDelayMinutes  \
0                             1447    HPN  TPA      -4.0              0.0   
1                             1451    BOS  MCO      53.0             53.0   
2                             1453    FLL  SJU       7.0              7.0   
3                             1454    SJU  FLL       7.0              7.0   
4                             1455    BOS  DCA      -8.0              0.0   

   DepDel15  ArrDelay  ArrDelayMinutes  Cancelled  Diverted  CarrierDelay  \
0       0.0      -4.0              0.0        0.0       0.0           NaN   
1       1.0      54.0             54.0        0.0       0.0          39.0   
2       0.0      23.0             23.0        0.0       0.0           7.0   
3       0.0       0.0              0.0        0.0       0.0           NaN   
4       0.0     -17.0              0.0        0.0       0.0           NaN   

   WeatherDelay  NASDelay  SecurityDelay  LateAircraftDelay  
0           NaN       NaN            NaN                NaN  
1           0.0       1.0            0.0               14.0  
2           0.0      16.0            0.0                0.0  
3           NaN       NaN            NaN                NaN  
4           NaN       NaN            NaN                NaN

In [5]:
# Empty and delete the extraction_path directory
try:
    shutil.rmtree(extraction_path)
    print(f"Directory {extraction_path} and all its contents have been deleted.")
except OSError as e:
    print(f"Error: {e.strerror}")

Directory Resources/BTS/extracted_zips and all its contents have been deleted.


## Airport Selection

In [6]:
# number of unique origin airports in data
origins_df = unfiltered_flight_info[['Origin','OriginCityName']]
origins_df['Origin'].nunique()

359

In [7]:
# number of unique destination airports in data
destinations_df = unfiltered_flight_info[['Dest','DestCityName']]
destinations_df['Dest'].nunique()

359

In [8]:
# total number of unique airports in data
# compare the origin airports to destination airports
ori_match = origins_df['Origin'].isin(destinations_df['Dest'])
dest_match = destinations_df['Dest'].isin(origins_df['Origin'])
ori_dest_match = ori_match | dest_match

# only keep rows for airports that are listed in both columns
flight_airports = flights_df[ori_dest_match]
flight_airports['Origin'].nunique()
# confirmed that the same 359 airports have travel data

359

## Above 3 cells combined and simplified below

In [30]:
# Determine total number of unique airports in flight data
origins = unfiltered_flight_info['Origin']
destinations = unfiltered_flight_info['Dest']
# compare the origin airports to destination airports
ori_match = origins.isin(destinations)
dest_match = destinations.isin(origins)
ori_dest_match = ori_match | dest_match

# only keep rows for airports that have inputs in both columns
flight_airports = flights_df[ori_dest_match]
# show number of unique airports that have travel data
flight_airports['Origin'].nunique()

359

In [9]:
#  compare us_airport data to airports in flight data
# Read in CSV of us-airports
us_airports = pd.read_csv('Resources/us-airports.csv')
# create exclusion
iata_match = us_airports['iata_code'].isin(flight_airports['Origin'])
# filter the data
us_airports_filtered = us_airports[iata_match].reset_index()
# count number of airports
us_airports_filtered['iata_code'].nunique()

351

In [10]:
# Which 8 airports do we have flight data for that our airport list doesn't have?
# create mask for inclusion
iata_match = flight_airports['Origin'].isin(us_airports_filtered['iata_code'])
# drop data that doesn't match 
us_airports_missing = flight_airports[~iata_match]
# show the extra airport codes in flight data
exclude_airports = us_airports_missing['Origin'].unique()
exclude_airports

array(['SJU', 'STT', 'BQN', 'PSE', 'STX', 'PPG', 'GUM', 'SPN'],
      dtype=object)

Researched IATA codes to determine location of airports(Google):
>SJU, BQM, PSE = Puerto Rico

>STT,STX, = U.S. Virgin Islands

>PPG = American Samoa (unincorporated territory)

>GUM = Guam

>SPN = Northern Mariana Islands

Conclusion: These 8 airports can be eliminated from flight data to keep it to 50 states.

Total airports: 351

In [11]:
# Excluding data from the 8 erroneous airports 
exclusions = flights_df['Origin'].isin(us_airports_missing['Origin'])

filtered_flights = flights_df[~exclusions]
filtered_flights['Origin'].nunique()

351

In [12]:
# Filter airports to just those that have flight data
airport_mask = us_airports['iata_code'].isin(filtered_flights['Origin'])

us_airports_filtered = pd.DataFrame(us_airports[airport_mask])
us_airports_filtered['iata_code'].nunique()

351

In [13]:
us_airports.head()

id        ident                 type  \
0  #meta +id  #meta +code  #loc +airport +type   
1       3632         KLAX        large_airport   
2       3754         KORD        large_airport   
3       3622         KJFK        large_airport   
4       3384         KATL        large_airport   

                                               name latitude_deg  \
0                               #loc +airport +name    #geo +lat   
1                 Los Angeles International Airport    33.942501   
2              Chicago O'Hare International Airport      41.9786   
3              John F Kennedy International Airport    40.639447   
4  Hartsfield Jackson Atlanta International Airport      33.6367   

  longitude_deg         elevation_ft                 continent  \
0     #geo +lon  #geo +elevation +ft  #region +continent +code   
1   -118.407997                  125                       NaN   
2      -87.9048                  680                       NaN   
3    -73.779317                   13                       NaN   
4    -84.428101                 1026                       NaN   

     country_name           iso_country  region_name        iso_region  \
0  #country +name  #country +code +iso2  #adm1 +name  #adm1 +code +iso   
1   United States                    US   California             US-CA   
2   United States                    US     Illinois             US-IL   
3   United States                    US     New York             US-NY   
4   United States                    US      Georgia             US-GA   

         local_region              municipality   scheduled_service  \
0  #adm1 +code +local  #loc +municipality +name  #status +scheduled   
1                  CA               Los Angeles                   1   
2                  IL                   Chicago                   1   
3                  NY                  New York                   1   
4                  GA                   Atlanta                   1   

                   gps_code                  iata_code  \
0  #loc +airport +code +gps  #loc +airport +code +iata   
1                      KLAX                        LAX   
2                      KORD                        ORD   
3                      KJFK                        JFK   
4                      KATL                        ATL   

                   local_code  \
0  #loc +airport +code +local   
1                         LAX   
2                         ORD   
3                         JFK   
4                         ATL   

                                           home_link  \
0                                #meta +url +airport   
1                            https://www.flylax.com/   
2  https://www.flychicago.com/ohare/home/pages/de...   
3                        https://www.jfkairport.com/   
4                    http://www.atlanta-airport.com/   

                                      wikipedia_link  \
0                              #meta +url +wikipedia   
1  https://en.wikipedia.org/wiki/Los_Angeles_Inte...   
2  https://en.wikipedia.org/wiki/O'Hare_Internati...   
3  https://en.wikipedia.org/wiki/John_F._Kennedy_...   
4  https://en.wikipedia.org/wiki/Hartsfield–Jacks...   

                                            keywords         score  \
0                                    #meta +keywords  #meta +score   
1                                        Tom Bradley       1335475   
2                                 CHI, Orchard Place       1503175   
3  Manhattan, New York City, NYC, Idlewild, IDL, ...       1052075   
4                                                NaN       2002475   

                last_updated  
0             #date +updated  
1  2023-12-21T12:31:02+00:00  
2  2024-03-09T23:28:49+00:00  
3  2022-10-18T18:49:55+00:00  
4  2018-09-19T14:50:01+00:00

In [14]:
us_airports_filtered.head()

id ident           type  \
1  3632  KLAX  large_airport   
2  3754  KORD  large_airport   
3  3622  KJFK  large_airport   
4  3384  KATL  large_airport   
5  3878  KSFO  large_airport   

                                               name       latitude_deg  \
1                 Los Angeles International Airport          33.942501   
2              Chicago O'Hare International Airport            41.9786   
3              John F Kennedy International Airport          40.639447   
4  Hartsfield Jackson Atlanta International Airport            33.6367   
5               San Francisco International Airport  37.61899948120117   

  longitude_deg elevation_ft continent   country_name iso_country region_name  \
1   -118.407997          125       NaN  United States          US  California   
2      -87.9048          680       NaN  United States          US    Illinois   
3    -73.779317           13       NaN  United States          US    New York   
4    -84.428101         1026       NaN  United States          US     Georgia   
5      -122.375           13       NaN  United States          US  California   

  iso_region local_region   municipality scheduled_service gps_code iata_code  \
1      US-CA           CA    Los Angeles                 1     KLAX       LAX   
2      US-IL           IL        Chicago                 1     KORD       ORD   
3      US-NY           NY       New York                 1     KJFK       JFK   
4      US-GA           GA        Atlanta                 1     KATL       ATL   
5      US-CA           CA  San Francisco                 1     KSFO       SFO   

  local_code                                          home_link  \
1        LAX                            https://www.flylax.com/   
2        ORD  https://www.flychicago.com/ohare/home/pages/de...   
3        JFK                        https://www.jfkairport.com/   
4        ATL                    http://www.atlanta-airport.com/   
5        SFO                             http://www.flysfo.com/   

                                      wikipedia_link  \
1  https://en.wikipedia.org/wiki/Los_Angeles_Inte...   
2  https://en.wikipedia.org/wiki/O'Hare_Internati...   
3  https://en.wikipedia.org/wiki/John_F._Kennedy_...   
4  https://en.wikipedia.org/wiki/Hartsfield–Jacks...   
5  https://en.wikipedia.org/wiki/San_Francisco_In...   

                                            keywords    score  \
1                                        Tom Bradley  1335475   
2                                 CHI, Orchard Place  1503175   
3  Manhattan, New York City, NYC, Idlewild, IDL, ...  1052075   
4                                                NaN  2002475   
5                                           QSF, QBA  1112475   

                last_updated  
1  2023-12-21T12:31:02+00:00  
2  2024-03-09T23:28:49+00:00  
3  2022-10-18T18:49:55+00:00  
4  2018-09-19T14:50:01+00:00  
5  2008-06-13T14:30:04+00:00

In [15]:
us_airports_filtered['type'].value_counts()

type
medium_airport    276
large_airport      65
small_airport      10
Name: count, dtype: int64

In [16]:
# Export filtered data sets to CSV
us_airports_filtered.to_csv('Resources/us_airports_filtered.csv')
filtered_flights.to_csv('Resources/filtered_flights.csv')

In [17]:
# # Zip filtered_flights.csv and delete to allow github upload
csv_path = 'Resources/filtered_flights.csv'
zip_path = 'Resources/filtered_flights.zip'

with zipfile.ZipFile(zip_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
    zipf.write(csv_path, os.path.basename(csv_path))

# Check if the ZIP file exists and the size condition is met
if os.path.exists(zip_path):
    # Get the size of the ZIP file in bytes
    zip_size = os.path.getsize(zip_path)
    
    # Define the size condition, e.g., file must be larger than 0 bytes to ensure it's not empty
    if zip_size > 50:
        os.remove(csv_path)
        print(f"The file {csv_path} has been zipped, is not empty, and has been deleted.")
    else:
        print(f"The file {zip_path} is empty.")
else:
    print("Error: The zip file was not created successfully.")

The file Resources/filtered_flights.csv has been zipped, is not empty, and has been deleted.


## Identify Airlines

In [18]:
# Identify unique IATA codes and store in a list in order of frequency
airline_IATA_codes = filtered_flights['IATA_Code_Marketing_Airline'].value_counts().index.to_list()
airline_IATA_codes

['AA', 'DL', 'WN', 'UA', 'AS', 'B6', 'NK', 'F9', 'G4', 'HA']

In [19]:
# Identify airlines and make list in matching order 
# Source: https://www.iata.org/en/publications/directories/code-search/
airline_names = ['American Airlines','Delta Airlines','Southwest Airlines','United Airlines',
                 'Alaska Airlines','JetBlue','Spirit Airlines','Frontier Airlines','Allegiant Air',
                 'Hawaiian Airlines']
# Build dictionary of IATA_codes and airline names to turn into a data frame
airline_data = {'airline_IATA_code':airline_IATA_codes,
               'airline_name':airline_names}

In [20]:
# create dataframe
airline_df = pd.DataFrame(airline_data)
airline_df

airline_IATA_code        airline_name
0                AA   American Airlines
1                DL      Delta Airlines
2                WN  Southwest Airlines
3                UA     United Airlines
4                AS     Alaska Airlines
5                B6             JetBlue
6                NK     Spirit Airlines
7                F9   Frontier Airlines
8                G4       Allegiant Air
9                HA   Hawaiian Airlines

In [21]:
# Export to CSV
airline_df.to_csv('Resources/airlines.csv')

## Get city of airport from the flight data and insert into airport information for SQL

In [22]:
us_airports_filtered.head(1)

id ident           type                               name latitude_deg  \
1  3632  KLAX  large_airport  Los Angeles International Airport    33.942501   

  longitude_deg elevation_ft continent   country_name iso_country region_name  \
1   -118.407997          125       NaN  United States          US  California   

  iso_region local_region municipality scheduled_service gps_code iata_code  \
1      US-CA           CA  Los Angeles                 1     KLAX       LAX   

  local_code                home_link  \
1        LAX  https://www.flylax.com/   

                                      wikipedia_link     keywords    score  \
1  https://en.wikipedia.org/wiki/Los_Angeles_Inte...  Tom Bradley  1335475   

                last_updated  
1  2023-12-21T12:31:02+00:00

In [23]:
# filter destination airports to unique entries
dest_filtered = destinations_df.drop_duplicates(subset=['Dest', 'DestCityName'])
dest_filtered.count()

Dest            359
DestCityName    359
dtype: int64

In [24]:
# check if there are any DestCityName rows that don't conform to City, State format
# regex = special characters include anything not a letter, number, comma, or whitespace
regex = r'^[A-Za-z0-9, ]+$'
# Dictionary to store results
not_uniform = []
# iterate through each row and check if cell contains comma AND doesn't match regex special char
for index, row in dest_filtered.iterrows():
    cell = row['DestCityName']
    if ',' in cell and not re.match(regex, cell):
        not_uniform.append(cell)

print(not_uniform)

# Learned: Even if there are slashes for metro areas, the format for ', state' still same

['West Palm Beach/Palm Beach, FL', 'Raleigh/Durham, NC', 'Sarasota/Bradenton, FL', 'Dallas/Fort Worth, TX', 'Montrose/Delta, CO', 'Bismarck/Mandan, ND', 'St. Louis, MO', 'Gulfport/Biloxi, MS', 'Jackson/Vicksburg, MS', 'Greensboro/High Point, NC', 'Cedar Rapids/Iowa City, IA', 'Jacksonville/Camp Lejeune, NC', 'Pasco/Kennewick/Richland, WA', 'Newburgh/Poughkeepsie, NY', 'Bloomington/Normal, IL', 'Allentown/Bethlehem/Easton, PA', 'St. Petersburg, FL', 'Bristol/Johnson City/Kingsport, TN', 'Elmira/Corning, NY', 'Mission/McAllen/Edinburg, TX', 'St. Cloud, MN', 'Clarksburg/Fairmont, WV', 'Manhattan/Ft. Riley, KS', 'Midland/Odessa, TX', 'Harlingen/San Benito, TX', 'Champaign/Urbana, IL', 'Scranton/Wilkes-Barre, PA', 'Charleston/Dunbar, WV', 'Ithaca/Cortland, NY', 'Saginaw/Bay City/Midland, MI', 'Newport News/Williamsburg, VA', 'New Bern/Morehead/Beaufort, NC', 'Lawton/Fort Sill, OK', 'Bend/Redmond, OR', 'Sun Valley/Hailey/Ketchum, ID', 'Beaumont/Port Arthur, TX', 'College Station/Bryan, TX', 

In [25]:
# Create airports DF with cities
# merge airports data with city data based on IATA code
airports_cities = us_airports_filtered.merge(dest_filtered, left_on='iata_code', 
                                            right_on='Dest', how ='inner')
# Get rid of the state abbreviation from cities column
airports_cities['city_region'] = airports_cities['DestCityName'].str.split(',').str[0]
# Get rid of _airport from 'type' column
airports_cities['airport_size'] = airports_cities['type'].str.split('_').str[0]
#reset index
airports_cities.reset_index(drop=True).head()
# select columns
airports_cities.head()

id ident           type  \
0  3632  KLAX  large_airport   
1  3754  KORD  large_airport   
2  3622  KJFK  large_airport   
3  3384  KATL  large_airport   
4  3878  KSFO  large_airport   

                                               name       latitude_deg  \
0                 Los Angeles International Airport          33.942501   
1              Chicago O'Hare International Airport            41.9786   
2              John F Kennedy International Airport          40.639447   
3  Hartsfield Jackson Atlanta International Airport            33.6367   
4               San Francisco International Airport  37.61899948120117   

  longitude_deg elevation_ft continent   country_name iso_country region_name  \
0   -118.407997          125       NaN  United States          US  California   
1      -87.9048          680       NaN  United States          US    Illinois   
2    -73.779317           13       NaN  United States          US    New York   
3    -84.428101         1026       NaN  United States          US     Georgia   
4      -122.375           13       NaN  United States          US  California   

  iso_region local_region   municipality scheduled_service gps_code iata_code  \
0      US-CA           CA    Los Angeles                 1     KLAX       LAX   
1      US-IL           IL        Chicago                 1     KORD       ORD   
2      US-NY           NY       New York                 1     KJFK       JFK   
3      US-GA           GA        Atlanta                 1     KATL       ATL   
4      US-CA           CA  San Francisco                 1     KSFO       SFO   

  local_code                                          home_link  \
0        LAX                            https://www.flylax.com/   
1        ORD  https://www.flychicago.com/ohare/home/pages/de...   
2        JFK                        https://www.jfkairport.com/   
3        ATL                    http://www.atlanta-airport.com/   
4        SFO                             http://www.flysfo.com/   

                                      wikipedia_link  \
0  https://en.wikipedia.org/wiki/Los_Angeles_Inte...   
1  https://en.wikipedia.org/wiki/O'Hare_Internati...   
2  https://en.wikipedia.org/wiki/John_F._Kennedy_...   
3  https://en.wikipedia.org/wiki/Hartsfield–Jacks...   
4  https://en.wikipedia.org/wiki/San_Francisco_In...   

                                            keywords    score  \
0                                        Tom Bradley  1335475   
1                                 CHI, Orchard Place  1503175   
2  Manhattan, New York City, NYC, Idlewild, IDL, ...  1052075   
3                                                NaN  2002475   
4                                           QSF, QBA  1112475   

                last_updated Dest       DestCityName    city_region  \
0  2023-12-21T12:31:02+00:00  LAX    Los Angeles, CA    Los Angeles   
1  2024-03-09T23:28:49+00:00  ORD        Chicago, IL        Chicago   
2  2022-10-18T18:49:55+00:00  JFK       New York, NY       New York   
3  2018-09-19T14:50:01+00:00  ATL        Atlanta, GA        Atlanta   
4  2008-06-13T14:30:04+00:00  SFO  San Francisco, CA  San Francisco   

  airport_size  
0        large  
1        large  
2        large  
3        large  
4        large

In [26]:
print(airports_cities['local_region'].unique())
print(airports_cities['region_name'].unique())

['CA' 'IL' 'NY' 'GA' 'NJ' 'TX' 'NV' 'FL' 'CO' 'VA' 'AZ' 'WA' 'MA' 'PA'
 'NC' 'DC' 'MI' 'MN' 'UT' 'MD' 'MO' 'HI' 'OR' 'LA' 'TN' 'KY' 'OH' 'IN'
 'NM' 'WI' 'CT' 'AK' 'OK' 'NE' 'RI' 'SC' 'AL' 'ID' 'KS' 'ME' 'IA' 'AR'
 'NH' 'VT' 'SD' 'WY' 'MT' 'ND' 'MS' 'WV']
['California' 'Illinois' 'New York' 'Georgia' 'New Jersey' 'Texas'
 'Nevada' 'Florida' 'Colorado' 'Virginia' 'Arizona' 'Washington'
 'Massachusetts' 'Pennsylvania' 'North Carolina' 'District of Columbia'
 'Michigan' 'Minnesota' 'Utah' 'Maryland' 'Missouri' 'Hawaii' 'Oregon'
 'Louisiana' 'Tennessee' 'Kentucky' 'Ohio' 'Indiana' 'New Mexico'
 'Wisconsin' 'Connecticut' 'Alaska' 'Oklahoma' 'Nebraska' 'Rhode Island'
 'South Carolina' 'Alabama' 'Idaho' 'Kansas' 'Maine' 'Iowa' 'Arkansas'
 'New Hampshire' 'Vermont' 'South Dakota' 'Wyoming' 'Montana'
 'North Dakota' 'Mississippi' 'West Virginia']


In [27]:
# drop and rename columns
airports_drop = airports_cities.drop(columns=['Dest','DestCityName','type'])
airports_df = airports_drop.rename(columns={'name':'airport_name',
                                               'region_name':'state',
                                              'local_region':'state_abbrv'})
airports_df.head()

id ident                                      airport_name  \
0  3632  KLAX                 Los Angeles International Airport   
1  3754  KORD              Chicago O'Hare International Airport   
2  3622  KJFK              John F Kennedy International Airport   
3  3384  KATL  Hartsfield Jackson Atlanta International Airport   
4  3878  KSFO               San Francisco International Airport   

        latitude_deg longitude_deg elevation_ft continent   country_name  \
0          33.942501   -118.407997          125       NaN  United States   
1            41.9786      -87.9048          680       NaN  United States   
2          40.639447    -73.779317           13       NaN  United States   
3            33.6367    -84.428101         1026       NaN  United States   
4  37.61899948120117      -122.375           13       NaN  United States   

  iso_country       state iso_region state_abbrv   municipality  \
0          US  California      US-CA          CA    Los Angeles   
1          US    Illinois      US-IL          IL        Chicago   
2          US    New York      US-NY          NY       New York   
3          US     Georgia      US-GA          GA        Atlanta   
4          US  California      US-CA          CA  San Francisco   

  scheduled_service gps_code iata_code local_code  \
0                 1     KLAX       LAX        LAX   
1                 1     KORD       ORD        ORD   
2                 1     KJFK       JFK        JFK   
3                 1     KATL       ATL        ATL   
4                 1     KSFO       SFO        SFO   

                                           home_link  \
0                            https://www.flylax.com/   
1  https://www.flychicago.com/ohare/home/pages/de...   
2                        https://www.jfkairport.com/   
3                    http://www.atlanta-airport.com/   
4                             http://www.flysfo.com/   

                                      wikipedia_link  \
0  https://en.wikipedia.org/wiki/Los_Angeles_Inte...   
1  https://en.wikipedia.org/wiki/O'Hare_Internati...   
2  https://en.wikipedia.org/wiki/John_F._Kennedy_...   
3  https://en.wikipedia.org/wiki/Hartsfield–Jacks...   
4  https://en.wikipedia.org/wiki/San_Francisco_In...   

                                            keywords    score  \
0                                        Tom Bradley  1335475   
1                                 CHI, Orchard Place  1503175   
2  Manhattan, New York City, NYC, Idlewild, IDL, ...  1052075   
3                                                NaN  2002475   
4                                           QSF, QBA  1112475   

                last_updated    city_region airport_size  
0  2023-12-21T12:31:02+00:00    Los Angeles        large  
1  2024-03-09T23:28:49+00:00        Chicago        large  
2  2022-10-18T18:49:55+00:00       New York        large  
3  2018-09-19T14:50:01+00:00        Atlanta        large  
4  2008-06-13T14:30:04+00:00  San Francisco        large

## Database Prep - dtypes, astype, rename

In [28]:
# Rather than small, medium, large being listed over and over - use 1,2,3 and have another table to classify

In [29]:
airports_df.dtypes

id                   object
ident                object
airport_name         object
latitude_deg         object
longitude_deg        object
elevation_ft         object
continent            object
country_name         object
iso_country          object
state                object
iso_region           object
state_abbrv          object
municipality         object
scheduled_service    object
gps_code             object
iata_code            object
local_code           object
home_link            object
wikipedia_link       object
keywords             object
score                object
last_updated         object
city_region          object
airport_size         object
dtype: object